client/src/pages/record.tsx

```diff
@@ -423,7 +424,7 @@ function RecordPage(props: {
               disabled={recordPageState === RecordPageState.RELOADING_MENTOR}
               onClick={() => {
                 setRecordPageState(RecordPageState.RELOADING_MENTOR);
-                saveSubject(recordState.reloadMentorData);
+                saveSubject().then(() => recordState.reloadMentorData());
```

Comment:
break this onClick handler out into a function with a name like `handleSaveSubject` (or whatever is appropriate)

As a rule of thumb, we only inline component event handlers if the code is a single statement